datajoint/declare.py

```diff
@@ -78,26 +78,53 @@ def compile_foreign_key(line, context, attributes, primary_key, attr_sql, foreig
     if not all(r in ref.primary_key for r in result.ref_attrs):
         raise DataJointError('Invalid foreign key attributes in "%s"' % line)
 
-    # match new attributes and referenced attributes and create foreign keys
-    missing_attrs = [attr for attr in ref.primary_key if attr not in attributes] or (
-        ref.primary_key if len(result.new_attrs) == len(ref.primary_key) == 1 else [])
-    new_attrs = result.new_attrs or missing_attrs
-    ref_attrs = result.ref_attrs or missing_attrs
+    try:
+        raise DataJointError('Duplicate attributes "{attr}" in "{line}"'.format(
+            attr=next(attr for attr in result.new_attrs if attr in attributes),
+            line=line))
+    except StopIteration:
+        pass   # the normal outcome
+
+    new_attrs = list(result.new_attrs)
+    ref_attrs = list(result.ref_attrs)
+
+    if new_attrs and not ref_attrs:
+        # special case, the renamed attribute is implicit
+        if len(new_attrs) != 1:
+            raise DataJointError('Renamed foreign key must be mapped to the primary key in "%s"' % line)
+        if len(ref.primary_key) == 1:
+            # if the primary key has one attribute, allow implicit renaming
+            ref_attrs = ref.primary_key
+        else:
+            # if only one primary key attribute is not yet used, then allow implicit renaming
```

Comment:
Should this case really be allowed?

It seems to encourage having attributes of the same name and also relies on sequence-of-definition which seems somewhat inconsistent with other properties of the DDL - though I perhaps could be misunderstanding the case/logic.

e.g.:

```
# 1st schema
@schema
class Foo(dj.Manual):
    definition = """
    fooid: integer
    barattr: integer
    """

@schema
class BarOne(dj.Manual):
    # is ok? (imports barattr as single attribute)
    definition = """
    fooid: integer
    -> Foo
    """

@schema
class BarTwo(dj.Manual):
    # is error: imports fooid and barattr; duplicate fooid results
    definition = """
    -> Foo
    fooid: integer
    """
```